DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,66 @@
+# Developer Guide
+
+So you want to contribute code to OpenSearch Go Client?  Excellent!  We're glad you're here.  Here's what you need to do:
+
+## Getting Started
+
+### Git Clone OpenSearch Go Client Repository
+
+Fork [opensearch-project/opensearch-go](https://github.com/opensearch-project/opensearch-go) and clone locally,
+e.g. `git clone https://github.com/[your username]/opensearch-go.git`.
+
+### Install Prerequisites
+
+#### Go 1.11
+OpenSearch Go Client builds using [Go](https://golang.org/doc/install) 1.11 at a minimum.
+
+#### Docker
+
+[Docker](https://docs.docker.com/install/) is required for building some OpenSearch artifacts and executing integration tests.
+
+### Unit Testing
+Go has a simple tool for running tests, and we simplified further by creating make command, use this command:
```

Comment:
nit: "simplified it further by creating this make command:"

Reply:
Ack